modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebarTreeView.tsx

```diff
@@ -0,0 +1,394 @@
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import {Text, TreeView} from '@clayui/core';
+import {ClayDropDownWithItems} from '@clayui/drop-down';
+import Icon from '@clayui/icon';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import {
+	API,
+	getLocalizableLabel,
+	stringIncludesQuery,
+} from '@liferay/object-js-components-web';
+import classNames from 'classnames';
+import {openToast, sub} from 'frontend-js-web';
+import React, {useMemo} from 'react';
+import {Node, useStore, useZoomPanHelper} from 'react-flow-renderer';
+
+import './LeftSidebar.scss';
+import {useObjectFolderContext} from '../ModelBuilderContext/objectFolderContext';
+import {TYPES} from '../ModelBuilderContext/typesEnum';
+import {LeftSidebarItem, LeftSidebarObjectDefinitionItem} from '../types';
+
+const TYPES_TO_SYMBOLS = {
+	linkedObjectDefinition: 'link',
+	objectDefinition: 'catalog',
+	objectFolder: 'folder',
+};
+
+export default function LeftSidebarTreeView({
+	query,
+	showActions,
+}: {
+	query: string;
+	showActions?: boolean;
+}) {
+	const [
+		{leftSidebarItems, selectedObjectFolder},
+		dispatch,
+	] = useObjectFolderContext();
+	const store = useStore();
+	const {setCenter} = useZoomPanHelper();
+
+	const changeObjectDefinitionNodeViewButton = (
+		hiddenObjectDefinitionNode: boolean,
+		dispatch: Function
+	) => (
+		<ClayButtonWithIcon
+			aria-label={
+				hiddenObjectDefinitionNode
+					? Liferay.Language.get('hidden')
+					: Liferay.Language.get('show')
+			}
+			displayType="unstyled"
+			onClick={(event) => {
+				event.stopPropagation();
+				dispatch();
+			}}
+			symbol={hiddenObjectDefinitionNode ? 'hidden' : 'view'}
+		/>
+	);
+
+	const filteredLeftSidebarItems = useMemo(() => {
+		return leftSidebarItems.map((sidebarItem) => {
+			if (!sidebarItem.leftSidebarObjectDefinitionItems) {
+				return sidebarItem;
+			}
+
+			const newObjectDefinitions = sidebarItem.leftSidebarObjectDefinitionItems.filter(
+				(leftSidebarObjectDefinitionItem) =>
+					stringIncludesQuery(
+						leftSidebarObjectDefinitionItem.label,
+						query
+					)
+			);
+
+			return {
+				...sidebarItem,
+				objectDefinitions: newObjectDefinitions,
+			};
+		});
+	}, [leftSidebarItems, query]);
+
+	const handleMove = async ({
+		objectDefinitionId,
+		objectFolderName,
+	}: {
+		objectDefinitionId: number;
+		objectFolderName: string;
+	}) => {
+		const objectFolders = await API.getAllObjectFolders();
+
+		const currentObjectFolder = objectFolders.find(
+			(objectFolder) => objectFolder.name === objectFolderName
+		);
+
+		const objectDefinitionsFilteredByObjectFolder = await API.getObjectDefinitions(
+			`filter=objectFolderExternalReferenceCode eq '${currentObjectFolder?.externalReferenceCode}'`
+		);
+
+		const objectDefinition = objectDefinitionsFilteredByObjectFolder.find(
+			(objectDefinition) => objectDefinition.id === objectDefinitionId
+		) as ObjectDefinitionNodeData;
```

Comment:
Here we are using the same name to constant and find item. So, if we created the pattern of the find and map names. Remeber the apply here.